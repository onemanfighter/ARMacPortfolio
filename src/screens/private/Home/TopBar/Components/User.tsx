import { Menu } from '@chakra-ui/react';
import { TopBarButton } from '@components';

const User = () => {
  return (
    <Menu>
      <TopBarButton
        text="Amit Raikwar"
        onClick={() => {}}
        ariaLabel="search-top-bar-button"
      />
    </Menu>
  );
};

export default User;
