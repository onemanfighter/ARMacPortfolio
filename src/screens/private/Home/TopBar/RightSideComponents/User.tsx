import { Box, Image, Menu, MenuDivider, Text } from '@chakra-ui/react';
import {
  MenuItemComponent,
  MenuListComponent,
  TopBarButton,
} from '@components';
import {
  darkModeColorSelector,
  settingsStore,
  useShallow,
  usersSelector,
} from '@settingsStore';
import { useTranslation } from 'react-i18next';

const User = () => {
  const { t } = useTranslation();
  const { textColor } = settingsStore(useShallow(darkModeColorSelector));
  const { name, profilePicture } = settingsStore(
    useShallow(usersSelector),
  ).userData;

  return (
    <Menu>
      <TopBarButton
        text={name}
        onClick={() => {}}
        ariaLabel="user-top-bar-button"
      />
      <MenuListComponent>
        <Box
          display={'flex'}
          width={'100%'}
          justifyContent={'center'}
          flexDirection={'column'}
          alignItems={'center'}
          py={2}
          color={textColor}
        >
          <Image
            borderRadius="full"
            src={profilePicture}
            alt="Profile"
            width="48px"
            loading="lazy"
            border={'2px solid white'}
          />
          <Text fontSize={11} fontWeight={600}>
            {name}
          </Text>
        </Box>
        <MenuDivider p={0} m={0.5} />
        <MenuItemComponent
          text={t('TopAppBar.user.loginWindow')}
          ariaLabel="login-window"
          onClick={() => {}}
        />
        <MenuDivider p={0} m={0.5} />
        <MenuItemComponent
          text={t('TopAppBar.user.setting')}
          ariaLabel="user-setting"
          onClick={() => {}}
        />
      </MenuListComponent>
    </Menu>
  );
};

export default User;
