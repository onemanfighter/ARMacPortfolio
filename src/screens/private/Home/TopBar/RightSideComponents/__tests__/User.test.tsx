import { fireEvent, render, screen } from '@testing-library/react';
import User from '../User';

describe('User', () => {
  it('should render', () => {
    const { container } = render(<User />);

    expect(container).toMatchSnapshot();
  });

  it('should open pop-up when User icon is tapped', () => {
    const { container } = render(<User />);

    fireEvent.click(screen.getByLabelText('user-top-bar-button'));

    expect(container).toMatchSnapshot();
  });

  it('should open user settings on user setting item clicked', () => {
    const { container } = render(<User />);

    fireEvent.click(screen.getByLabelText('user-setting'));

    expect(container).toMatchSnapshot();
  });

  it('should open login window on login window item clicked', () => {
    const { container } = render(<User />);

    fireEvent.click(screen.getByLabelText('login-window'));

    expect(container).toMatchSnapshot();
  });
});
