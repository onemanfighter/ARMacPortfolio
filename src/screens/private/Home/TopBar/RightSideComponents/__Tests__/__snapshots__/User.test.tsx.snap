// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should open login window on login window item clicked 1`] = `
<div>
  <button
    aria-controls="menu-list-:rd:"
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="user-top-bar-button"
    class="chakra-menu__menu-button css-10l0d22"
    id="menu-button-:rd:"
  >
    <span
      class="css-xl71ch"
    >
      <p
        class="chakra-text css-0"
      >
        Amit Raikwar
      </p>
    </span>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-f6jf1x"
      id="menu-list-:rd:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="css-1hdcak2"
      >
        <img
          alt="Profile"
          class="chakra-image css-1khlcnb"
          loading="lazy"
          src="https://avatars.githubusercontent.com/u/104697219?s=400&u=7ecc539c268755cfe0409ca5863773bb726387ba&v=4"
        />
        <p
          class="chakra-text css-gsdmxf"
        >
          Amit Raikwar
        </p>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="login-window"
        class="chakra-menu__menuitem css-n7rl60"
        data-index="0"
        id="menu-list-:rd:-menuitem-:re:"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        Login window...
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="user-setting"
        class="chakra-menu__menuitem css-n7rl60"
        data-index="1"
        id="menu-list-:rd:-menuitem-:rf:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Users & group settings...
      </button>
    </div>
  </div>
</div>
`;

exports[`User should open pop-up when User icon is tapped 1`] = `
<div>
  <button
    aria-controls="menu-list-:r5:"
    aria-expanded="true"
    aria-haspopup="menu"
    aria-label="user-top-bar-button"
    class="chakra-menu__menu-button css-10l0d22"
    data-active=""
    id="menu-button-:r5:"
  >
    <span
      class="css-xl71ch"
    >
      <p
        class="chakra-text css-0"
      >
        Amit Raikwar
      </p>
    </span>
  </button>
  <div
    class="css-0"
    data-popper-escaped=""
    data-popper-placement="bottom-start"
    data-popper-reference-hidden=""
    style="visibility: visible; position: absolute; min-width: max-content; inset: 0 auto auto 0; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 8px); --popper-transform-origin: top left;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-f6jf1x"
      id="menu-list-:r5:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="css-1hdcak2"
      >
        <img
          alt="Profile"
          class="chakra-image css-1khlcnb"
          loading="lazy"
          src="https://avatars.githubusercontent.com/u/104697219?s=400&u=7ecc539c268755cfe0409ca5863773bb726387ba&v=4"
        />
        <p
          class="chakra-text css-gsdmxf"
        >
          Amit Raikwar
        </p>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="login-window"
        class="chakra-menu__menuitem css-n7rl60"
        data-index="0"
        id="menu-list-:r5:-menuitem-:r6:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Login window...
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="user-setting"
        class="chakra-menu__menuitem css-n7rl60"
        data-index="1"
        id="menu-list-:r5:-menuitem-:r7:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Users & group settings...
      </button>
    </div>
  </div>
</div>
`;

exports[`User should open user settings on user setting item clicked 1`] = `
<div>
  <button
    aria-controls="menu-list-:r9:"
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="user-top-bar-button"
    class="chakra-menu__menu-button css-10l0d22"
    id="menu-button-:r9:"
  >
    <span
      class="css-xl71ch"
    >
      <p
        class="chakra-text css-0"
      >
        Amit Raikwar
      </p>
    </span>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-f6jf1x"
      id="menu-list-:r9:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="css-1hdcak2"
      >
        <img
          alt="Profile"
          class="chakra-image css-1khlcnb"
          loading="lazy"
          src="https://avatars.githubusercontent.com/u/104697219?s=400&u=7ecc539c268755cfe0409ca5863773bb726387ba&v=4"
        />
        <p
          class="chakra-text css-gsdmxf"
        >
          Amit Raikwar
        </p>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="login-window"
        class="chakra-menu__menuitem css-n7rl60"
        data-index="0"
        id="menu-list-:r9:-menuitem-:ra:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Login window...
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="user-setting"
        class="chakra-menu__menuitem css-n7rl60"
        data-index="1"
        id="menu-list-:r9:-menuitem-:rb:"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        Users & group settings...
      </button>
    </div>
  </div>
</div>
`;

exports[`User should render 1`] = `
<div>
  <button
    aria-controls="menu-list-:r1:"
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="user-top-bar-button"
    class="chakra-menu__menu-button css-10l0d22"
    id="menu-button-:r1:"
  >
    <span
      class="css-xl71ch"
    >
      <p
        class="chakra-text css-0"
      >
        Amit Raikwar
      </p>
    </span>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-f6jf1x"
      id="menu-list-:r1:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="css-1hdcak2"
      >
        <img
          alt="Profile"
          class="chakra-image css-1khlcnb"
          loading="lazy"
          src="https://avatars.githubusercontent.com/u/104697219?s=400&u=7ecc539c268755cfe0409ca5863773bb726387ba&v=4"
        />
        <p
          class="chakra-text css-gsdmxf"
        >
          Amit Raikwar
        </p>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="login-window"
        class="chakra-menu__menuitem css-n7rl60"
        data-index="0"
        id="menu-list-:r1:-menuitem-:r2:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Login window...
      </button>
      <hr
        aria-orientation="horizontal"
        class="chakra-menu__divider css-1fmk8qx"
      />
      <button
        aria-label="user-setting"
        class="chakra-menu__menuitem css-n7rl60"
        data-index="1"
        id="menu-list-:r1:-menuitem-:r3:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Users & group settings...
      </button>
    </div>
  </div>
</div>
`;
